android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/DefaultAddonsProvider.kt

```diff
@@ -42,23 +42,23 @@ internal const val DEFAULT_READ_TIMEOUT_IN_SECONDS = 20L
 internal const val PAGE_SIZE = 50
 
 /**
- * Provide access to the AMO collections API.
- * https://addons-server.readthedocs.io/en/latest/topics/api/collections.html
+ * Implement a default add-ons provider using the AMO API.
  *
  * @property context A reference to the application context.
  * @property client A [Client] for interacting with the AMO HTTP api.
  * @property serverURL The url of the endpoint to interact with e.g production, staging
  * or testing. Defaults to [DEFAULT_SERVER_URL].
  * @property collectionUser The id or name of the user owning the collection specified in
- * [collectionName], defaults to [DEFAULT_COLLECTION_USER].
- * @property collectionName The name of the collection to access, defaults
- * to [DEFAULT_COLLECTION_NAME].
- * @property maxCacheAgeInMinutes maximum time (in minutes) the collection cache
- * should remain valid before a refresh is attempted. Defaults to -1, meaning no
- * cache is being used by default
+ * [collectionName], defaults to [DEFAULT_COLLECTION_USER]. This is used to retrieve the
+ * featured add-ons.
+ * @property collectionName The name of the collection to access, defaults to
+ * [DEFAULT_COLLECTION_NAME]. This is used to retrieve the featured add-ons.
+ * @property maxCacheAgeInMinutes maximum time (in minutes) the cached featured add-ons
+ * should remain valid before a refresh is attempted. Defaults to -1, meaning no cache
+ * is being used by default
  */
 @Suppress("LongParameterList")
-class AddonCollectionProvider(
+class DefaultAddonsProvider(
```

Comment:
Could you expand on why should we rename? I think `AddonCollectionProvider` stills convey the idea that is generic class with give you access to the the AMO collections API, not sure if `DefaultAddonsProvider` provides the same message for other consumers.

Reply:
I didn't like `AMOAddonsProvider` (though I could probably be live with `AMOProvider`).  A new method is added in #3157, which will call the AMO search API, hence why the name can no longer be "collection" something something.

Reply:
Thanks for the extra context, that helped to clarify!

I feel you `AMOAddonsProvider` is not perfect, but for now it conveys its main purpose goal, for now I think it does the job 😅